network/validator/pubsub/authorized_sender_validator_test.go

```diff
@@ -0,0 +1,293 @@
+			s.Require().NoError(err)
+
+			validate := AuthorizedSenderValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+
+			msgType, err := validate(ctx, pid, c.Message)
+			s.Require().ErrorIs(err, message.ErrUnauthorizedMessageOnChannel)
+			s.Require().Equal(c.MessageStr, msgType)
+
+			validatePubsub := AuthorizedSenderMessageValidator(zerolog.Nop(), c.Channel, c.GetIdentity)
+			pubsubResult := validatePubsub(ctx, pid, c.Message)
+			s.Require().Equal(pubsub.ValidationReject, pubsubResult)
+		})
+	}
+}
+
+// TestValidatorCallback_ClusterPrefixedChannels checks that the call back returned from AuthorizedSenderValidator correctly
+// handles cluster prefixed channels during validation.
+func (s *TestAuthorizedSenderValidatorSuite) TestValidatorCallback_ClusterPrefixedChannels() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	identity, _ := unittest.IdentityWithNetworkingKeyFixture(unittest.WithRole(flow.RoleCollection))
+	clusterID := flow.Localnet
+
+	getIdentityFunc := s.getIdentity(identity)
+	pid, err := unittest.PeerIDFromFlowID(identity)
+	s.Require().NoError(err)
+
+	// validate collection consensus cluster
+	validateCollConsensus := AuthorizedSenderValidator(zerolog.Nop(), channels.ConsensusCluster(clusterID), getIdentityFunc)
+	msgType, err := validateCollConsensus(ctx, pid, &messages.ClusterBlockResponse{})
+	s.Require().NoError(err)
+	s.Require().Equal(message.ClusterBlockResponse, msgType)
+
+	validateCollConsensusPubsub := AuthorizedSenderMessageValidator(zerolog.Nop(), channels.ConsensusCluster(clusterID), getIdentityFunc)
+	pubsubResult := validateCollConsensusPubsub(ctx, pid, &messages.ClusterBlockResponse{})
+	s.Require().Equal(pubsub.ValidationAccept, pubsubResult)
+
+	// validate collection sync cluster
+	validateSyncCluster := AuthorizedSenderValidator(zerolog.Nop(), channels.SyncCluster(clusterID), getIdentityFunc)
+	msgType, err = validateSyncCluster(ctx, pid, &messages.SyncRequest{})
+	s.Require().NoError(err)
+	s.Require().Equal(message.SyncRequest, msgType)
+
+	validateSyncClusterPubsub := AuthorizedSenderMessageValidator(zerolog.Nop(), channels.SyncCluster(clusterID), getIdentityFunc)
+	pubsubResult = validateSyncClusterPubsub(ctx, pid, &messages.SyncRequest{})
+	s.Require().Equal(pubsub.ValidationAccept, pubsubResult)
+}
+
+// TestValidatorCallback_ValidationFailure checks that the call back returned from AuthorizedSenderValidator returns the expected validation error.
+func (s *TestAuthorizedSenderValidatorSuite) TestValidatorCallback_ValidationFailure() {
+	s.Run("sender is ejected", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		identity, _ := unittest.IdentityWithNetworkingKeyFixture()
+		identity.Ejected = true
+		getIdentityFunc := s.getIdentity(identity)
+		pid, err := unittest.PeerIDFromFlowID(identity)
+		s.Require().NoError(err)
+
+		validate := AuthorizedSenderValidator(zerolog.Nop(), channels.SyncCommittee, getIdentityFunc)
```

Comment:
A general comment: instead of using the `Nop()` logger, please use the `unittest.Logger()`. It is acting as `Nop` by default, however, it enables logging when passing the `-vv` flag, which is very helpful in the event of debugging our tests.